### PR TITLE
Solving issue #5623 and supporting bools for C client label:"Client: C"

### PR DESCRIPTION
**Pull request by michelealbano**
- It adds support for Booleans.
- It solves issue #5623 
- Remove stray printfs that were left around during a previous PR.
- It solves #5616 

In particular, for #5623 I had to allow to enter `{{^isPrimitiveType}}` only when I am into `{{^isListContainer}}`, for the api header and body of C's mustache files.

@zhemant @ityuhui 

### PR checklist

- [ X ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ X ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [ X ] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [ X ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [ X] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/C-libcurl/model-body.mustache [hunk 1]
```diff
@@ -389,12 +394,18 @@ cJSON *{{classname}}_convertToJSON({{classname}}_t *{{classname}}) {
...
     }
     {{/isString}}
-    {{^isString}}
+    {{#isBoolean}}
+    if(cJSON_AddBoolToObject({{{name}}}, "", *(cJSON_bool *){{{name}}}ListEntry->data) == NULL)
+    {
+        goto fail;
+    }
+    {{/isBoolean}}
+    {{^isNumeric}}
```
Comment (ityuhui) [inline modules/openapi-generator/src/main/resources/C-libcurl/model-body.mustache]: I think this shoud be  {{#isNumeric}}

File: modules/openapi-generator/src/main/resources/C-libcurl/model-body.mustache [hunk 2]
```diff
@@ -433,12 +444,19 @@ cJSON *{{classname}}_convertToJSON({{classname}}_t *{{classname}}) {
...
-        {{^isString}}
+        {{#isNumeric}}
         if(cJSON_AddNumberToObject(localMapObject, localKeyValue->key, *(double *)localKeyValue->value) == NULL)
         {
             goto fail;
         }
-        {{/isString}}
+        {{/isNumeric}}
+        {{#isBoolean}}
+//         if(cJSON_AddBoolToObject(localMapObject, localKeyValue->key, (cJSON_bool)localKeyValue->value) == NULL)
```
Comment (ityuhui) [inline modules/openapi-generator/src/main/resources/C-libcurl/model-body.mustache]: Could you please remove this comment ?

File: modules/openapi-generator/src/main/resources/C-libcurl/model-body.mustache (hunk 1 above)
Comment (michelealbano) [reply modules/openapi-generator/src/main/resources/C-libcurl/model-body.mustache]: Yes!

File: modules/openapi-generator/src/main/resources/C-libcurl/model-body.mustache (hunk 2 above)
Comment (michelealbano) [reply modules/openapi-generator/src/main/resources/C-libcurl/model-body.mustache]: Done

File: modules/openapi-generator/src/main/resources/C-libcurl/api-body.mustache [hunk 3]
```diff
@@ -99,7 +99,8 @@ end:
...
+{{#isModel}}{{#isEnum}}{{datatypeWithEnum}}_e{{/isEnum}}{{^isEnum}}{{{dataType}}}_t *{{/isEnum}}{{/isModel}}{{^isModel}}{{#isEnum}}{{datatypeWithEnum}}_e{{/isEnum}}{{/isModel}}{{#isUuid}}{{dataType}} *{{/isUuid}}{{#isEmail}}{{dataType}}{{/isEmail}}{{/isPrimitiveType}}{{/isListContainer}}{{#isContainer}}{{#isListContainer}}{{dataType}}_t *{{/isListContainer}}{{#isMapContainer}}{{dataType}}{{/isMapContainer}}{{/isContainer}} {{{paramName}}} {{/allParams}})
```
Comment (ityuhui) [inline modules/openapi-generator/src/main/resources/C-libcurl/api-body.mustache]: Could you please concatenate the two lines (102,103) to one line, just like previous ?

File: modules/openapi-generator/src/main/resources/C-libcurl/api-header.mustache [hunk 4]
```diff
@@ -32,7 +32,8 @@ typedef enum  { {{projectName}}_{{operationId}}_{{enumName}}_NULL = 0{{#enumVars
...
+{{#isModel}}{{#isEnum}}{{datatypeWithEnum}}_e{{/isEnum}}{{^isEnum}}{{{dataType}}}_t *{{/isEnum}}{{/isModel}}{{^isModel}}{{#isEnum}}{{datatypeWithEnum}}_e{{/isEnum}}{{/isModel}}{{#isUuid}}{{dataType}} *{{/isUuid}}{{#isEmail}}{{dataType}}{{/isEmail}}{{/isPrimitiveType}}{{/isListContainer}}{{#isContainer}}{{#isListContainer}}{{dataType}}_t *{{/isListContainer}}{{#isMapContainer}}{{dataType}}{{/isMapContainer}}{{/isContainer}} {{{paramName}}} {{/allParams}});
```
Comment (ityuhui) [inline modules/openapi-generator/src/main/resources/C-libcurl/api-header.mustache]: Could you please concatenate the two lines (102,103) to one line, just like previous ?

File: modules/openapi-generator/src/main/resources/C-libcurl/api-body.mustache (hunk 3 above)
Comment (michelealbano) [reply modules/openapi-generator/src/main/resources/C-libcurl/api-body.mustache]: Done

File: samples/client/petstore/c/model/pet.c [hunk 5]
```diff
@@ -110,6 +110,10 @@ cJSON *pet_convertToJSON(pet_t *pet) {
     {
         goto fail;
     }
+    if(cJSON_AddNumberToObject(photo_urls, "", *(double *)photo_urlsListEntry->data) == NULL)
+    {
+        goto fail;
+    }
```
Comment (ityuhui) [inline samples/client/petstore/c/model/pet.c]: This sample code is not regenerated by the latest template code.

File: modules/openapi-generator/src/main/resources/C-libcurl/api-header.mustache (hunk 4 above)
Comment (michelealbano) [reply modules/openapi-generator/src/main/resources/C-libcurl/api-header.mustache]: Done

File: samples/client/petstore/c/model/pet.c (hunk 5 above)
Comment (michelealbano) [reply samples/client/petstore/c/model/pet.c]: Done, and pushed to github

**Comment by ityuhui** [comment]
LGTM

**Comment by michelealbano** [comment]
@zhemant and @wing328 any comment on this PR?

**Comment by zhemant** [comment]
The PR is good

**Comment by michelealbano** [comment]
@wing328 it seems that this PR is ready to merge

**Pull request merged**